boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -37,74 +52,103 @@ void App_SevenSegDisplays_Destroy(struct SevenSegDisplays *const seven_seg_displ
     free(seven_seg_displays);
 }
 
-ExitCode App_SevenSegDisplays_SetHexDigits(
-    const struct SevenSegDisplays *const seven_seg_displays,
-    const uint8_t                        hex_digits[],
-    size_t                               num_hex_digits)
+void App_SevenSegDisplays_SetDigits(const struct SevenSegDisplays *seven_seg_displays, uint8_t group_index)
 {
-    if (num_hex_digits > NUM_SEVEN_SEG_DISPLAYS || num_hex_digits == 0)
+    for (int i = group_index; i < NUM_IN_GROUP + group_index; i++)
+    {
+        struct SevenSegHexDigit hex_digit;
+        hex_digit.enabled = true;
+        hex_digit.value   = digits[i];
+
+        App_SevenSegDisplay_SetHexDigit(seven_seg_displays->displays[i], hex_digit, i);
+    }
+
+    seven_seg_displays->display_value_callback();
+}
+
+ExitCode App_SevenSegDisplays_SetGroupL(const struct SevenSegDisplays *const seven_seg_displays, uint32_t value)
+{
+    if (value > pow(10, NUM_IN_GROUP) - 1)
     {
         return EXIT_CODE_INVALID_ARGS;
     }
+    digits[LEFT_L_SEVEN_SEG_DISPLAY] = 0;
+    digits[LEFT_M_SEVEN_SEG_DISPLAY] = 0;
+    digits[LEFT_R_SEVEN_SEG_DISPLAY] = 0;
 
-    // If any of the input digits is invalid, we don't write anything to the
-    // 7-segment displays at all.
-    for (size_t i = 0; i < num_hex_digits; i++)
+    // Turn the base-10 value into individual digits. Have to write backwards with how the
+    // displays are initialized and how they are passed to the IO function.
+    for (int digits_index = LEFT_R_SEVEN_SEG_DISPLAY; digits_index + LEFT_R_SEVEN_SEG_DISPLAY > NUM_IN_GROUP; digits_index--)
     {
-        if (hex_digits[i] >= NUM_HEX_DIGITS)
+        digits[digits_index - 1] = (uint8_t)(value % 10);
+        value /= 10;
+
+        if (value == 0)
         {
-            return EXIT_CODE_INVALID_ARGS;
+            // initializing from the first place it needs to edit, due to the inverse nature of the daisy chain in the IO
+            App_SevenSegDisplays_SetDigits(seven_seg_displays, 0);
+            break;
         }
     }
 
-    for (size_t i = 0; i < NUM_SEVEN_SEG_DISPLAYS; i++)
+    return EXIT_CODE_OK;
+}
+
+ExitCode App_SevenSegDisplays_SetGroupM(const struct SevenSegDisplays *const seven_seg_displays, uint32_t value)
+{
+    if (value > pow(10, NUM_IN_GROUP) - 1)
     {
-        struct SevenSegHexDigit hex_digit;
+        return EXIT_CODE_INVALID_ARGS;
+    }
+    digits[MIDDLE_L_SEVEN_SEG_DISPLAY] = 0;
+    digits[MIDDLE_M_SEVEN_SEG_DISPLAY] = 0;
+    digits[MIDDLE_R_SEVEN_SEG_DISPLAY] = 0;
 
-        if (i < num_hex_digits)
-        {
-            hex_digit.enabled = true;
-            hex_digit.value   = hex_digits[i];
-        }
-        else
+    // Turn the base-10 value into individual digits. Have to write backwards with how the
+    // displays are initialized and how they are passed to the IO function
+    for (int digits_index = MIDDLE_R_SEVEN_SEG_DISPLAY; digits_index + MIDDLE_R_SEVEN_SEG_DISPLAY > NUM_IN_GROUP; digits_index--)
```

Comment:
Regarding code style, might be more human-readable to directly index the `digits` array using the ints 0-8, and simply write which index of the array corresponds to which location on the display. 

For example: SetGroupL might look like
for (i = 0; i < 3; i++) // do stuff to digits[i]

Whereas SetGroupM might be 
for (i = 3; i < 6; i++) // do stuff to digits[i]

Reply:
I'm usually against unnamed constants but I can see the enums here is unclear
Gonna leave this one up to Ashli